app/routes/ffxiv.shopping-list.tsx

```diff
@@ -24,28 +25,38 @@ import { getItemIDByName } from '~/utils/items'
 import { ffxivItemsList } from '~/utils/items/id_to_item'
 
 const FORM_DEFAULTS = {
-  craft_amount: 1,
+  craft_amount: 5,
   hq: true,
-  job: 8
+  job: 0
 }
 
 const FORM_NAME = 'shopping-list'
+const REGION_WIDE = 'region-wide'
 
 type ShoppingFormItem = ShoppingInputItem & { name: string }
 type ActionDataResponse = GetShoppingListResponse | { exception: string } | {}
 
 export const action: ActionFunction = async ({ request }) => {
-  const formData = (await request.formData()).get(FORM_NAME)
-  if (!formData || typeof formData !== 'string') {
+  const formData = await request.formData()
+
+  const shoppingData = formData.get(FORM_NAME)
+  const regionData = formData.get(REGION_WIDE)
+
+  if (!shoppingData || typeof shoppingData !== 'string') {
     return json({ exception: 'Input not found' })
   }
+
   try {
-    const shoppingList = JSON.parse(formData)
+    const shoppingList = JSON.parse(shoppingData)
     const session = await getUserSessionData(request)
 
     const homeServer = session.getWorld()
 
-    return GetShoppingList({ shoppingList, homeServer })
+    return GetShoppingList({
+      shoppingList,
+      homeServer,
+      regionWide: regionData === 'on'
+    })
   } catch {
     return json({ exception: 'Invalid form data.' })
   }
```

Comment:
When we call `regionData === 'on'` it is converting the value into a boolean, if regionData is null, then the check will return false, if regionData is undefined it will return false, if regionData is an empty object it will return false, if regionData is not the string 'on' it will return false. So whether or not the value is a string, it will always be converted into a boolean.
Is there any time this check could error?

Reply:
@coderabbitai can you answer my question:
Is there a time when the check `regionData === 'on'` would error?